src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -432,6 +433,20 @@ public boolean authenticate(final SecurityRequestChannel request) {
         return authenticated;
     }
 
+    /**
+     * Checks if incoming auth request is from an anonymous user
+     * Defaults all requests to yes, to allow anonymous authentication to succeed
+     * @param params the query parameters passed in this request
+     * @return false only if an explicit `auth_request_type` param is supplied and its value is not anonymous,
+     * otherwise returns true
+     */
+    private boolean isRequestForAnonymousLogin(Map<String, String> params) {
+        if (params.containsKey("auth_request_type")) {
```

Comment:
This is a new public interface - we wouldn't be able to change it after releasing the feature, lets be careful about how we name it, as non-dashboard clients can use this behavior too.

- When using non standard http headers they are preceded with an X
- Header keys are kebab cased
- the word request is implied making it redundant.

What do you think about `X-Auth-Type`?

Reply:
@peternied FYI its not required to pass this url param and any request is assumed anonymous unless its explicitly said to not be.

i.e. 

- `curl -XGET https://localhost:9200 -k` is assumed to be anonymous and will give 200 response
- `curl -XGET https://localhost:9200?auth_request_type=saml -k` is explicitly passed to tell the backend auth registry to not allow anonymous login with the request and will give back a 401 Forbidden response if the request is not properly authed.

Reply:
it is no longer a header. It is a request param. I agree that request is redundant here, so i have rename it to `auth_type`. My understanding is that request params are snake_case. Let me know if otherwise.